org.scala-ide.sdt.debug/plugin.xml

```diff
@@ -28,7 +28,7 @@
             delegate="org.scalaide.debug.internal.launching.ScalaApplicationLaunchConfigurationDelegate"
             delegateDescription="The Scala JVM Launcher supports debugging of local Scala using the new Scala debugger"
             id="scala.application.new"
-            modes="debug"
+            modes="debug, run"
```

Comment:
It is a little bit defensive. scalatest can be run both in run and debug
modes. By now its launcher is using default vm runner (with
AbstractJavaLaunchConfigurationDelegate). I wanted to change the hierarchy
to use ScalaApplicationLaunchConfigurationDelegate. But it makes a trouble
because defined runner is a type of debugger so in run mode it stops in
breakpoints.
Tbh I didn't delve to deep into the matter to check what happens when I
define one launch configuration type with modes set to run, debug and one
launch delegate of this type with modes set just to 'debug'. I'll check it.

2015-05-27 15:00 GMT+02:00 Simon Schäfer <notifications@github.com>:

> In org.scala-ide.sdt.debug/plugin.xml
> <https://github.com/scala-ide/scala-ide/pull/955#discussion_r31130133>:
>
> > @@ -28,7 +28,7 @@
> >              delegate="org.scalaide.debug.internal.launching.ScalaApplicationLaunchConfigurationDelegate"
> >              delegateDescription="The Scala JVM Launcher supports debugging of local Scala using the new Scala debugger"
> >              id="scala.application.new"
> > -            modes="debug"
> > +            modes="debug, run"
>
> I don't yet understand why we need this change. Want we be able to run the
> EE in both debug and run mode or is ScalaTest only available in run mode
> but somehow executed in debug mode?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scala-ide/scala-ide/pull/955/files#r31130133>.
>